samples/matter/thermostat/src/temperature_measurement/sensor.cpp

```diff
@@ -20,31 +21,80 @@ using namespace ::chip;
 using namespace ::chip::DeviceLayer;
 using namespace ::chip::app;
 
-LOG_MODULE_DECLARE(app);
+LOG_MODULE_DECLARE(app, CONFIG_CHIP_APP_LOG_LEVEL);
 
 namespace
 {
 
 constexpr chip::EndpointId mTemperatureMeasurementEndpointId = 1;
 
+constexpr uint16_t kSimulatedReadingFrequency = (60000 / kSensorTimerPeriodMs); /*Change simulated number every minute*/
+
+#if CONFIG_THERMOSTAT_TEMPERATURE_STEP == 0
+constexpr int16_t sMockTemp[] = { 2000, 2731, 1600, 2100, 1937, 3011, 1500, 1899 };
+#endif
+
+
 } // namespace
 
-RealSensor::RealSensor()
+TemperatureSensor::TemperatureSensor()
 {
 	BindingHandler::Init();
 }
 
-void RealSensor::TemperatureMeasurement()
+void TemperatureSensor::FullMeasurement(){
+	InternalMeasurement();
+	ExternalMeasurement();
+}
+
+void TemperatureSensor::InternalMeasurement()
+{
+#if CONFIG_THERMOSTAT_TEMPERATURE_STEP == 0
+
+	int16_t temperature = sMockTemp[mMockTempIndex];
+
+	mCycleCounter++;
+	if (mCycleCounter >= kSimulatedReadingFrequency) {
+		if (mMockTempIndex >= ArraySize(sMockTemp) - 1) {
+			mMockTempIndex = 0;
+		} else {
+			mMockTempIndex++;
+		}
+		mCycleCounter = 0;
+	}
+#else
+#ifdef CONFIG_THERMOSTAT_SIMULATED_TEMPERATURE_CHANGE
+	static int16_t temperature = CONFIG_THERMOSTAT_SIMULATED_TEMPERATURE_MIN - CONFIG_THERMOSTAT_TEMPERATURE_STEP;
+
+	temperature = (PreviousTemperature >= CONFIG_THERMOSTAT_SIMULATED_TEMPERATURE_MAX) ?
```

Comment:
Should it be `mPreviousTemperature`? Could you verify that it builds with non-zero step as well?